softlayer/resource_softlayer_global_ip.go

```diff
@@ -0,0 +1,279 @@
+	AdditionalServicesGlobalIpAddressesPackageType = "ADDITIONAL_SERVICES_GLOBAL_IP_ADDRESSES"
+
+	GlobalIpMask = "id,ipAddress[ipAddress],destinationIpAddress[ipAddress]"
+)
+
+func resourceSoftLayerGlobalIp() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceSoftLayerGlobalIpCreate,
+		Read:     resourceSoftLayerGlobalIpRead,
+		Update:   resourceSoftLayerGlobalIpUpdate,
+		Delete:   resourceSoftLayerGlobalIpDelete,
+		Exists:   resourceSoftLayerGlobalIpExists,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"ip_address": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"routes_to": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceSoftLayerGlobalIpCreate(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(*session.Session)
+
+	// Find price items with AdditionalServicesGlobalIpAddresses
+	productOrderContainer, err := buildGlobalIpProductOrderContainer(d, sess, AdditionalServicesGlobalIpAddressesPackageType)
+	if err != nil {
+		// Find price items with AdditionalServices
+		productOrderContainer, err = buildGlobalIpProductOrderContainer(d, sess, AdditionalServicesPackageType)
+		if err != nil {
+			return fmt.Errorf("Error creating global ip: %s", err)
+		}
+	}
+
+	log.Println("[INFO] Creating global ip")
+
+	receipt, err := services.GetProductOrderService(sess).
+		PlaceOrder(productOrderContainer, sl.Bool(false))
+	if err != nil {
+		return fmt.Errorf("Error during creation of global ip: %s", err)
+	}
+
+	globalIp, err := findGlobalIpByOrderId(sess, *receipt.OrderId)
+	d.SetId(fmt.Sprintf("%d", *globalIp.Id))
+	return resourceSoftLayerGlobalIpUpdate(d, meta)
+
+}
+
+func resourceSoftLayerGlobalIpRead(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(*session.Session)
+	service := services.GetNetworkSubnetIpAddressGlobalService(sess)
+
+	globalIpId, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return fmt.Errorf("Not a valid global ip ID, must be an integer: %s", err)
+	}
+
+	globalIp, err := service.Id(globalIpId).Mask(GlobalIpMask).GetObject()
+	if err != nil {
+		if apiErr, ok := err.(sl.Error); ok && apiErr.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving Global Ip: %s", err)
+	}
+
+	d.Set("id", *globalIp.Id)
+	d.Set("ip_address", *globalIp.IpAddress.IpAddress)
+	d.Set("routes_to", *globalIp.DestinationIpAddress.IpAddress)
+
+	return nil
+}
+
+func resourceSoftLayerGlobalIpUpdate(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(*session.Session)
+	service := services.GetNetworkSubnetIpAddressGlobalService(sess)
+
+	globalIpId, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return fmt.Errorf("Not a valid global ip ID, must be an integer: %s", err)
+	}
+
+	if d.HasChange("routes_to") {
```

Comment:
Since _Update_ is called from _Create_, I think this condition will be false that first call (`d.HasHcange("routes_to")` will be false). So either pull out the code within the condition as a helper function that can be called from _Create_ without any interruptions and can allso be called from here (_Update_). Or, remove the `if d.HasChange("routes_to")` conditional, since *routes_to* is the only updatable field, so _Update_ won't be called unless it has changed.